corehq/apps/locations/models.py

```diff
@@ -399,35 +397,58 @@ class SQLLocation(SyncSQLToCouchMixin, MPTTModel):
     active_objects = OnlyUnarchivedLocationManager()
 
     @classmethod
-    def _migration_get_fields(cls):
+    def get_sync_fields(cls):
         return ["domain", "name", "site_code", "external_id",
                 "metadata", "is_archived"]
 
-    @classmethod
-    def _migration_get_couch_model_class(cls):
-        return Location
+    def sync_couch_location(self):
+        if self.location_id:
+            couch_obj = self.couch_location
+        else:
+            couch_obj = Location()
+            couch_obj._id = uuid.uuid4().hex
+            self.location_id = couch_obj._id
```

Comment:
:+1: Agreed - to be clear, this is setting a random uuid instead of hitting the db.  There's also some info here related to performance based on how ids are generated: http://docs.couchdb.org/en/1.6.1/maintenance/performance.html?highlight=id#document-s-id

Right now we just do random ids which doesn't seem like the best approach, but generating our own sequential ids might increase the chances of conflicts unless we find a good way to do it.